eth2near/utilities/src/serde.rs

```diff
@@ -0,0 +1,86 @@
+use num_traits::Num;
+use serde::{de, Deserialize, Deserializer};
+use crate::primitives::{Byte, Bytes, FixedBytes};
+
+pub fn hex_to_int_opt<'de, D, N>(deserializer: D) -> Result<Option<N>, D::Error>
+    where D: Deserializer<'de>,
+          N: Num,
+{
+    Ok(Some(hex_to_int(deserializer)?))
+}
+
+pub fn hex_to_int<'de, D, N>(deserializer: D) -> Result<N, D::Error>
```

Comment:
Some of the ser/de are implemented here https://docs.rs/ethereum_serde_utils/latest/serde_utils/
So please take a look into it and try to reuse them if possible, like here https://github.com/aurora-is-near/rainbow-bridge/blob/6a2e37708ec5a72cc0dc99b7634f0cc796ea9278/contracts/near/eth-types/src/lib.rs#L186-L187

Reply:
I discovered that this crate doesn't handle odd length strings. If a hex string has leading zeros ETH RPCs send the shortened version.
My code has handling for this (although now I wonder that when there are 2+ leading zeros my code may also fail so I need to handle it)

Reply:
Ethereum spec showing first character cannot be a zero https://github.com/ethereum/execution-apis/blob/main/src/schemas/base-types.yaml#L58

Reply:
Removed some duplication and added tests in 97bc7648291bc25f4e88625f4e589b51c3344209